### PR TITLE
Refactor function names and fix installing 7zip locally if already globally available

### PR DESCRIPTION
**Pull request by r15ch13**
Fixes:
- Prevents installing 7zip/lessmsi/innounp locally if they are already available globally

Additions:
- `Get-HelperPath`
- `Test-HelperInstalled`
- ~`Get-LessmsiPath`~ `Get-HelperPath -Helper Lessmsi`
- ~`Test-LessmsiInstalled`~ `Test-HelperInstalled -Helper Lessmsi`
- ~`Get-InnounpPath`~ `Get-HelperPath -Helper Innounp`
- ~`Test-InnounpInstalled`~ `Test-HelperInstalled -Helper Innounp`
- `Get-HelperPath -Helper Dark`
- `Test-HelperInstalled -Helper Dark`

Added tests for:
- `Get-AppFilePath`
- `Get-HelperPath`
- `Test-HelperInstalled`
- `Test-Aria2Enabled`
- `Test-CommandAvailable`

Refactors:
- `aria2_path` > `Get-HelperPath -Helper Aria2`
- `aria2_installed` > `Test-HelperInstalled -Helper Aria2`
- `aria2_enabled` > `Test-Aria2Enabled`
- `7zip_path` > `Get-HelperPath -Helper 7zip`
- `7zip_installed` > `Test-HelperInstalled -Helper 7zip`
- `Expand-7ZipArchive` > `Expand-7zipArchive`
- `Test-7ZipRequirement` > `Test-7zipRequirement`
- `Expand-MSIArchive` > `Expand-MsiArchive`
- `Test-LessMSIRequirement` > `Test-LessmsiRequirement`
- `file_path` > `Get-AppFilePath`

### CODE REVIEW

**Comment by r15ch13** [comment]
@Ash258 and @niheaven

**Comment by niheaven** [comment]
Consider `Test-CommandAvailable`, and will we think about external installation of these tools? That is, test if they are installed (internal or external), call with `file_path`, if error fallback to cmdname (`7z.exe`, `lessmsi.exe`, `innounp.exe`, like `7zip_using_external` do).

PS. I could test it tomorrow~

**Comment by r15ch13** [comment]
Since these are essential for Scoop to work, I would rather not use external versions because these can cause issues. Older 7zip versions have already caused some issues.

**Comment by niheaven** [comment]
Okay, got it.

Since `Get-xxxPath` return filepath string, it could substitute those string in `Expand-xxx` function, couldn't it? i.e. `& Get-7zipPath x "$Path" -o"$DestinationPath"`, will it works?

**Comment by niheaven** [comment]
```powershell
& (Get-7zipPath) x "$Path" -o"$DestinationPath" (-split $Switches) -y | Out-File $LogLocation
```

Above cmd works, and I suggest change them so, for further scoop refactor.

File: lib/core.ps1 [hunk 1]
```diff
@@ -152,16 +152,31 @@ function installed_apps($global) {
...
+        [String]
+        $File
+    )
+
     # normal path to file
-    $path = "$(versiondir $app 'current' $false)\$file"
-    if(Test-Path($path)) {
-        return $path
+    $Path = "$(versiondir $App 'current' $false)\$File"
+    if(Test-Path($Path)) {
```
Comment (niheaven) [inline lib/core.ps1]:
```suggestion
    if(Test-Path $Path) {
```

File: lib/core.ps1 [hunk 2]
```diff
@@ -152,16 +152,31 @@ function installed_apps($global) {
...
+        return $Path
     }
 
     # global path to file
-    $path = "$(versiondir $app 'current' $true)\$file"
-    if(Test-Path($path)) {
-        return $path
+    $Path = "$(versiondir $App 'current' $true)\$File"
+    if(Test-Path($Path)) {
```
Comment (niheaven) [inline lib/core.ps1]:
```suggestion
    if(Test-Path $Path) {
```

File: lib/core.ps1 [hunk 3]
```diff
@@ -175,24 +190,38 @@ Function Test-CommandAvailable {
...
+    )
 
-function aria2_installed() {
-    return ![String]::IsNullOrWhiteSpace("$(aria2_path)")
+    return ![String]::IsNullOrWhiteSpace((Get-HelperPath -Helper $Helper))
 }
 
-function aria2_enabled() {
-    return (aria2_installed) -and (get_config 'aria2-enabled' $true)
+function Test-Aria2Enabled() {
```
Comment (niheaven) [inline lib/core.ps1]:
```suggestion
function Test-Aria2Enabled {
```

**Pull request merged**